persistence.js

```diff
@@ -525,42 +543,48 @@ RedisPersistence.prototype.delWill = function (client, cb) {
 }
 
 RedisPersistence.prototype.streamWill = function (brokers) {
-  return new MatchStream({
-    objectMode: true,
-    redis: this._db,
-    match: 'will:*',
-    count: 100
-  })
-  .pipe(through.obj(function streamWill (chunk, enc, cb) {
-    for (var i = 0, l = chunk.length; i < l; i++) {
-      if (!brokers || !brokers[chunk[i].split(':')[1]]) {
-        this.push(chunk[i])
+  var stream = throughv.obj(this._decodeAndAugment)
+
+  this._db.lrange(willKey, 0, 10000, streamWill)
+
+  function streamWill (err, results) {
+    if (err) {
+      stream.emit('error', err)
+    } else {
+      for (var i = 0, l = results.length; i < l; i++) {
+        if (!brokers || !brokers[results[i].split(':')[1]]) {
+          stream.write(results[i])
+        }
       }
+      stream.end()
     }
-    cb()
-  }))
-  .pipe(throughv.obj(this._decodeAndAugment))
+  }
+  return stream
 }
 
 RedisPersistence.prototype.getClientList = function (topic) {
   var that = this
+  var key = 'sub:client:' + topic
 
-  return new MatchStream({
-    objectMode: true,
-    redis: this._db,
-    match: 'sub:client:' + topic
-  })
-  .pipe(through.obj(function getStream (chunk, enc, cb) {
+  var stream = through.obj(getStream)
+
+  stream.write(key)
+  stream.end()
```

Comment:
we do not need this stream at all, we can just write the result of `hgetall` to the through stream.